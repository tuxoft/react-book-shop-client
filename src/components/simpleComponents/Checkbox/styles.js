import styled from "styled-components";

export const CheckboxWrapper = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row;
`;
export const CheckboxImage = styled.img`
  width: 16px;
  height: 16px;
  padding-right: 12px;
`;
