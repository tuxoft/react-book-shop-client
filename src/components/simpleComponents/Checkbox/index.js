import React, {Component} from "react";
import * as styles from "./styles";
import PropTypes from "prop-types";
import {SimpleText} from "../../simpleComponents/Text";

const checkboxNonChecked = require("../../../assets/images/checkbox.png");
const checkboxChecked = require("../../../assets/images/checkbox-selected.png");


const CheckboxComponent = ({isChecked, onClick, labelText}) =>(
    <styles.CheckboxWrapper onClick = {() =>{onClick(!isChecked)}}>  {/* нужно инвертирование, т.к. state предыдущий*/}
        <styles.CheckboxImage src = {isChecked ? checkboxChecked : checkboxNonChecked}/>
        {labelText && <SimpleText>{labelText}</SimpleText>}
    </styles.CheckboxWrapper>
);


class Checkbox extends  Component {
    constructor (props){
       super(props);
       this.state = {
           isChecked: false
       }
    }
    toggleChecked(){
        this.setState({
            isChecked: !this.state.isChecked
        });
    }
    render (){
        return(<CheckboxComponent {...this.state}
                                  onClick = {(e)=> {
                                      this.props.onClick(e);
                                      this.toggleChecked();
                                  }}
                                  labelText = {this.props.labelText}/>)
    }
}
Checkbox.propTypes = {
    onClick: PropTypes.func,
    labelText:PropTypes.string
};
export default Checkbox;
