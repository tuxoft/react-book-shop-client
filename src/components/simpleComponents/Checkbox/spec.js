import React from "react";
import Checkbox from "./index";
import { shallow, mount } from "enzyme";

it("checkbox", () => {
  const onClick = jest.fn();
  const cb = shallow(<Checkbox labelText={'text'} onClick={onClick}/>);
  expect(cb).toMatchSnapshot();
});
it("checkbox handler", () => {
     let expected = true;
     const onClickHandler =(val) =>{
         expect(val).toBe(expected);
     };
    const cb = mount(<Checkbox labelText={'text'} onClick={onClickHandler}/>);
    cb.simulate('click', {target:{}});
    expected = false;
    cb.simulate('click', {target:{}});

});