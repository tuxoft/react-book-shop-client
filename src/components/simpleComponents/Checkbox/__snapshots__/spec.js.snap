// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox 1`] = `
<CheckboxComponent
  isChecked={false}
  labelText="text"
  onClick={[Function]}
/>
`;
